apis/python/src/tiledbsoma/_tdb_handles.py

```diff
@@ -209,21 +197,8 @@ def _opener(
         uri: str,
         mode: options.OpenMode,
         context: SOMATileDBContext,
-        use_latest_read_timestamp: bool = False,
     ) -> tiledb.Group:
-        if not use_latest_read_timestamp:
-            # As of Feb 2023, tiledb.Group() has no timestamp arg; instead its timestamps must be
-            # set in the tiledb.Ctx config. SOMATileDBContext prepares the suitable tiledb.Ctx.
-            ctx_arg = (
-                context._group_write_tiledb_ctx
-                if mode == "w"
-                else context._group_read_tiledb_ctx
-            )
-        else:
-            # Group opened in write mode should initialize with latest contents & metadata
```

Comment:
I think this is kind of a thorny question, if for whatever reason we've opened the object for writing at a *past* timestamp (preceding other, existing, later-timestamped writes), what view of the contents/metadata should we start with? But that's a weird enough thing to be doing in the first place that I'm not too worried about it. Simplifying the code is good.